pkg/kn/commands/source/apiserver/flags.go

```diff
@@ -41,27 +42,45 @@ type APIServerSourceUpdateFlags struct {
 }
 
 type resourceSpec struct {
-	kind         string
-	apiVersion   string
-	isController bool
+	Kind         string
+	ApiVersion   string
+	IsController bool
 }
 
-// GetAPIServerResourceArray is to return an array of ApiServerResource from a string. A sample is Event:v1:true,Pod:v2:false
-func (f *APIServerSourceUpdateFlags) GetAPIServerResourceArray() (*[]v1alpha1.ApiServerResource, error) {
-	var resourceList []v1alpha1.ApiServerResource
+// getAPIServerResourceArray is to return an array of ApiServerResource from a string. A sample is Event:v1:true,Pod:v2:false
+func (f *APIServerSourceUpdateFlags) getAPIServerResourceArray() ([]resourceSpec, error) {
+	var resourceList []resourceSpec
 	for _, r := range f.Resources {
 		resourceSpec, err := getValidResource(r)
 		if err != nil {
 			return nil, err
 		}
-		resourceRef := v1alpha1.ApiServerResource{
-			APIVersion: resourceSpec.apiVersion,
-			Kind:       resourceSpec.kind,
-			Controller: resourceSpec.isController,
+		resourceList = append(resourceList, *resourceSpec)
+	}
+	return resourceList, nil
+}
+
+// getAPIServerResourceArray is to return an array of ApiServerResource from a string. A sample is Event:v1:true
+func (f *APIServerSourceUpdateFlags) getUpdateAPIServerResourceArray() ([]resourceSpec, []resourceSpec, error) {
+	var added []resourceSpec
+	var removed []resourceSpec
+
+	addedArray, removedArray := util.AddListAndRemovalListFromArray(f.Resources)
+	for _, r := range addedArray {
+		resourceSpec, err := getValidResource(r)
+		if err != nil {
+			return nil, nil, err
+		}
+		added = append(added, *resourceSpec)
+	}
+	for _, r := range removedArray {
+		resourceSpec, err := getValidResource(r)
+		if err != nil {
+			return nil, nil, err
 		}
-		resourceList = append(resourceList, resourceRef)
+		removed = append(removed, *resourceSpec)
 	}
-	return &resourceList, nil
+	return added, removed, nil
 }
 
 func getValidResource(resource string) (*resourceSpec, error) {
```

Comment:
you should add comment for all code. I would rather see for public methods and functions but since you added above then should be consistent...

Reply:
Added a comment.